src/components/App/App.js

```diff
@@ -159,10 +159,16 @@ const App = () => {
       updateUser(user);
 
       Promise.all([
-        user && getFavorites().then(setFavorites),
+        user &&
+        getFavorites().then(setFavorites),
         getMentors().then(setMentors),
       ]).then(() => {
         setIsReady(true);
+        readAndUpdateFavMentorsFromLocalStorage().then((isMentorListUpdated) => {
```

Comment:
If we already have the favorites list, from localStorage,  this function can return that and call `setFavorites` with the result (and save the call to the server).

Reply:
I have made the suggested changes in the latest commit.